src/jit/MemoryInl.h

```diff
@@ -322,10 +355,56 @@ static void emitLoad(sljit_compiler* compiler, Instruction* instr)
     sljit_u32 size;
     sljit_u32 offset = 0;
     uint32_t options = 0;
+    sljit_u16 memIndex = 0;
 #ifdef HAS_SIMD
     sljit_s32 simdType = 0;
 #endif /* HAS_SIMD */
 
+    switch (instr->opcode()) {
```

Comment:
Option: you could add a flag to info which represent that the opcode has an index. It would simplify this check